src/core/public/http/http_service.ts

```diff
@@ -17,84 +17,26 @@
  * under the License.
  */
 
-import * as Rx from 'rxjs';
-import {
-  distinctUntilChanged,
-  endWith,
-  map,
-  pairwise,
-  startWith,
-  takeUntil,
-  tap,
-} from 'rxjs/operators';
-
-import { Deps } from './types';
-import { setup } from './fetch';
+import { HttpDeps, HttpSetup, HttpStart, HttpServiceBase } from './types';
+import { setup } from './http_setup';
 
 /** @internal */
 export class HttpService {
-  private readonly loadingCount$ = new Rx.BehaviorSubject(0);
-  private readonly stop$ = new Rx.Subject();
-
-  public setup(deps: Deps) {
-    const { fetch, shorthand } = setup(deps);
-
-    return {
-      fetch,
-      delete: shorthand('DELETE'),
-      get: shorthand('GET'),
-      head: shorthand('HEAD'),
-      options: shorthand('OPTIONS'),
-      patch: shorthand('PATCH'),
-      post: shorthand('POST'),
-      put: shorthand('PUT'),
-      addLoadingCount: (count$: Rx.Observable<number>) => {
-        count$
-          .pipe(
-            distinctUntilChanged(),
-
-            tap(count => {
-              if (count < 0) {
-                throw new Error(
-                  'Observables passed to loadingCount.add() must only emit positive numbers'
-                );
-              }
-            }),
+  private service!: HttpServiceBase;
```

Comment:
if you assert type manually https://github.com/elastic/kibana/pull/36611/files#diff-efcab20ad7e4167fb45155ba26879300R34 you don't need `!`, but `?`